speech/speech_analytics_openai.py

```diff
@@ -0,0 +1,114 @@
+# /// script
+# description = "Summarize podcasts"
+# requires-python = ">=3.10, <3.13"
+# dependencies = ["daft[lance]", "openai", "python-dotenv", "soundfile", "numpy"]
+# ///
+import io
+import pathlib
+
+import daft
+from daft import DataType
+from openai import OpenAI
+import soundfile as sf
+
+
+@daft.func(
+    return_dtype=DataType.struct(
+        {
+            "transcript": DataType.string(),
+            "segments": DataType.list(
+                DataType.struct(
+                    {
+                        "seg_text": DataType.string(),
+                        "seg_start": DataType.float32(),
+                        "seg_end": DataType.float32(),
+                    }
+                )
+            ),
+        }
+    )
+)
+def transcribe(audio_file: daft.File):
+    """Transcribes an audio file using openai whisper."""
+
+    client = OpenAI()
+
+    file_obj = io.BytesIO(audio_file.read())
+    file_obj.name = "audio.mp3"
```

Comment:
so @kevinzwang just reminded me that we made file immutable. You need to use a context manager to open it now


```suggestion
    with file.open() as f:
        file_obj = io.BytesIO(f.read())
        file_obj.name = "audio.mp3"
```